### PR TITLE
Add dispatch for `provider_ehr_launch` based on `queryparams`

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
The body of the function remains unchanged just added an additional dispatch on `queryparams` dict

### CODE REVIEW

**Comment by andreasnoack** [comment]
We noticed that it's only the parameters that are actually used in the function body and it would be simpler for us to just pass the parameters. I'm wondering if it would be even better to just pass `iss` and `launch_token` directly as arguments instead of passing the Dict.

File: src/provider/ehr_launch.jl [hunk 1]
```diff
@@ -20,7 +20,7 @@ function provider_ehr_launch(config::ProviderEHRLaunchConfig,
                              uri_string::AbstractString;
                              kwargs...)
     uri = URIs.URI(uri_string)
-    return provider_ehr_launch(config, uri; kwargs...)
+    return provider_ehr_launch(config, queryparams(uri); kwargs...)
```
Comment (DilumAluthge) [inline src/provider/ehr_launch.jl:23]:
```suggestion
    return provider_ehr_launch(config, URIs.queryparams(uri); kwargs...)
``'

File: src/provider/ehr_launch.jl [hunk 2]
```diff
@@ -32,7 +33,17 @@ end
...
+end
+"""
+    provider_ehr_launch(config::ProviderEHRLaunchConfig, queryparams::Dict; kwargs...)
+
+## Optional Keyword Arguments:
+- `scope::AbstractString`. Default value: `"launch"`.
+"""
+function provider_ehr_launch(config::ProviderEHRLaunchConfig,
+                             queryparams::Dict;
```
Comment (DilumAluthge) [inline src/provider/ehr_launch.jl:45]:
```suggestion
                             queryparams::Dict{String,String};
```

**Review by DilumAluthge — APPROVED** [review]

File: src/provider/ehr_launch.jl (hunk 1 above)
Comment (DilumAluthge) [inline src/provider/ehr_launch.jl:23]:
```suggestion
    return provider_ehr_launch(config, URIs.queryparams(uri); kwargs...)
```

**Review by DilumAluthge — APPROVED** [review]

File: src/provider/ehr_launch.jl [hunk 3]
```diff
@@ -32,7 +32,17 @@ end
 function provider_ehr_launch(config::ProviderEHRLaunchConfig,
                              uri::URIs.URI;
                              scope::AbstractString = "launch")
-    queryparams = URIs.queryparams(uri)
+    return provider_ehr_launch(config, queryparams(uri); kwargs...)
```
Comment (DilumAluthge) [inline src/provider/ehr_launch.jl:35]:
```suggestion
    return provider_ehr_launch(config, URIs.queryparams(uri); kwargs...)
```

**Review by DilumAluthge — APPROVED** [review]

**Comment by DilumAluthge** [comment]
> I'm wondering if it would be even better to just pass `iss` and `launch_token` directly as arguments instead of passing the Dict.

- #23

**Pull request merged**